### PR TITLE
 REFACT - PDF Module error messages

### PR DESCRIPTION
**Pull request by carlwilson**
- added markers for all error numbers;
- reviewed and documented errors (in spreadsheet for wiki);
- added a few new constants in place of magic numbers;
- refactoring and simplification for:
  - Destinations
  - DocNode
  - FileSpecification
  - PdfModule
  - PageObject
  - PdfStream
- removed unthrown exceptions in method definitions;
- removed unused variables and code;
- changed some private method names to be more descriptive of function;
- explicit types for generic declarations;
- removed associated and unnecessary casts; and
- added missing `@Override` annotations.

### CODE REVIEW

File: jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java [hunk 1]
```diff
@@ -1550,7 +1533,7 @@ else if ("f".equals(keyval)) {
                         }
                         else {
                             throw new PdfMalformedException
-                                   (MessageConstants.ERR_XREF_TABLE_OPERATOR_ILLEGAL,
+                                   (MessageConstants.ERR_XREF_TABLE_OPERATOR_ILLEGAL, // PDF-HUL-83
```
Comment (david-russo) [inline jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java]: PDF-HUL-84

File: jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java [hunk 2]
```diff
@@ -831,7 +815,7 @@ public final void parse(RandomAccessFile raf, RepInfo info)
             }
             // Beware infinite loop on badly broken file
             if (_startxref == _prevxref) {
-				info.setMessage(new ErrorMessage(MessageConstants.ERR_XREF_TABLES_BROKEN,
+				info.setMessage(new ErrorMessage(MessageConstants.ERR_XREF_TABLES_BROKEN, // {PDF-HUL-134
```
Comment (david-russo) [inline jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java]: Extra "{"

File: jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java [hunk 3]
```diff
@@ -1597,22 +1580,22 @@ protected boolean readDocCatalogDict(RepInfo info)
...
         }
         try {
             // Check that the catalog has a key type and the types value is "Catalog"
             if (!checkTypeKey(_docCatDict, info, KEY_VAL_CATALOG,
-                              MessageConstants.ERR_DOC_CAT_TYPE_INVALID,
-                              MessageConstants.ERR_DOC_CAT_NO_TYPE,
-                              MessageConstants.ERR_DOC_CAT_NOT_SIMPLE)) {
+                              MessageConstants.ERR_DOC_CAT_TYPE_INVALID, // PDF-HUL-141
+                              MessageConstants.ERR_DOC_CAT_NO_TYPE, // PDF-HUL-142
+                              MessageConstants.ERR_DOC_CAT_NOT_SIMPLE)) { // PDF-HUL-142
```
Comment (david-russo) [inline jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java]: PDF-HUL-143?

File: jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java [hunk 4]
```diff
@@ -3021,7 +2962,7 @@ protected Property buildPageProperty(PageObject page,
                         // but we do report that we don't report them.
                         if (!_skippedAnnotationsReported) {
                             info.setMessage(new InfoMessage
-                                (annotationsSkippedString));
+                                (MessageConstants.INF_ANNOTATIONS_SKIPPED));
```
Comment (david-russo) [inline jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java]: PDF-HUL-115

File: jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java [hunk 5]
```diff
@@ -2943,7 +2884,7 @@ protected void addPagesProperty(List<Property> metadataList, RepInfo info)
             else {
                 if (!_skippedPagesReported) {
                     info.setMessage(new InfoMessage
-                       (pagesSkippedString));
+                       (MessageConstants.INF_PAGES_SKIPPED));
```
Comment (david-russo) [inline jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java]: PDF-HUL-112

File: jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/Parser.java [hunk 6]
```diff
@@ -96,7 +94,7 @@ public Token getNext (long max) throws IOException, PdfException
         else if (tok instanceof DictionaryEnd) {
             --_dictDepth;
             if (_dictDepth < 0) {
-                throw new PdfMalformedException (MessageConstants.ERR_DICT_DELIMETERS_IMPROPERLY_NESTED);
+                throw new PdfMalformedException (MessageConstants.ERR_DICT_DELIMITERS_IMPROPERLY_NESTED);
```
Comment (david-russo) [inline jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/Parser.java]: PDF-HUL-33

Comment (carlwilson) [reply jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/Parser.java]: Good catch, am thinking of adding a consistency check for errors in unit tests, but this is why adding an error number to the message soon is a good idea. Then the compiler takes care of it.

**Pull request merged**